cdc/puller/unified_sorter.go

```diff
@@ -0,0 +1,686 @@
+package puller
+
+import (
+	"bufio"
+	"container/heap"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"go.uber.org/zap"
+	"io"
+	"math"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+const (
+	fileBufferSize     = 16 * 1024 * 1024
+	heapSizeLimit      = 4 * 1024 * 1024 // 4MB
+	numConcurrentHeaps = 16
+	memoryLimit        = 1024 * 1024 * 1024 // 1GB
```

Comment:
Could you set the value base on system memory capacity, eg. min(upperBound, x% * systemMemory).